File: src/main/java/ch/uzh/ifi/access/model/dto/AssignmentProgressDTO.java
package ch.uzh.ifi.access.model.dto;

import ch.uzh.ifi.access.projections.EvaluationSummary;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Data
@RequiredArgsConstructor
@AllArgsConstructor
public class AssignmentProgressDTO {

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private String userId;
    @NonNull
    private String url;
    @NonNull
    private List<EvaluationSummary> tasks;
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/AssignmentProgressDTO.kt
package ch.uzh.ifi.access.model.dto

import ch.uzh.ifi.access.model.AssignmentInformation
import ch.uzh.ifi.access.projections.AssignmentInformationPublic
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_EMPTY)
class AssignmentProgressDTO(
    val userId: String? = null,
    val slug: String? = null,
    val information: MutableMap<String, AssignmentInformationDTO> = mutableMapOf(),
    val tasks: List<TaskProgressDTO>? = null
)

